pym/portage/sync/modules/git/git.py

```diff
@@ -133,20 +148,29 @@ def update(self):
 		exitcode = portage.process.spawn_bash("cd %s ; exec %s" % (
 				portage._shell_quote(self.repo.location), git_cmd),
 			**self.spawn_kwargs)
+
+		if exitcode == os.EX_OK:
+			if not self.verify_head(revision='refs/remotes/%s^..' % remote_branch):
+				return (1, False)
+
+			merge_cmd = [self.bin_command, 'merge', 'refs/remotes/%s' % remote_branch]
+			if quiet:
+				merge_cmd.append('--quiet')
+			exitcode = subprocess.call(merge_cmd,
+				cwd=portage._unicode_encode(self.repo.location))
+
 		if exitcode != os.EX_OK:
 			msg = "!!! git pull error in %s" % self.repo.location
```

Comment:
Maybe this should be changed to `msg = "!!! git fetch error ...`

Reply:
Ok, updated to show separate `fetch` and `merge` error messages.